### DISCUSSION TITLE
"Unrar: The specified password is incorrect." from nzbget v24.3 in my Debian stable (bookworm).

### DISCUSSION THREAD

**Discussion by antdude** [nzbgetcom/nzbget#397] [Q&A] [answered]
Hello,

Through a text SSH2 PuTTY connection, I downloaded and ran https://github.com/nzbgetcom/nzbget/releases/download/v24.3/nzbget-24.3-bin-linux.run in my updated 64-bit Debian stable (bookworm)'s ~/Downloads, moved ~/Downloads/nzbget to ~/bin, and configured its nzbget.conf's basic settings like MainDir=~/Downloads/nzbget (can't figure out how to make GitHub not cross out my texts -- see my https://www.reddit.com/r/nzbget/comments/1fnd3vu/unrar_the_specified_password_is_incorrect_from/ for a clearer post), news.newshosting.com, logins, etc. I ran ~/bin/nzbget/nzbget -o outputmode=color F.nzb, but it kept saying the rar files have invalid passwords like:
```
...
Sun Sep 22 22:05:51 2024        INFO    Successfully downloaded F/F.part04.rar
Sun Sep 22 22:05:51 2024        INFO    Directly unpacking F
Sun Sep 22 22:05:51 2024        DETAIL  Unpacking into /home/ant/Downloads/nzbget/completed/F/_unpack
Sun Sep 22 22:05:51 2024        INFO    Unrar: UNRAR 5.70 freeware      Copyright (c) 1993-2019 Alexander Roshal
Sun Sep 22 22:05:51 2024        INFO    Unrar: The specified password is incorrect.
Sun Sep 22 22:05:51 2024        WARNING Cancelling direct unpack for F
Sun Sep 22 22:05:51 2024        WARNING Interrupted direct unpack for F
Sun Sep 22 22:05:51 2024        WARNING Direct unpack for F failed
...

```
I tried nzbget's unrar7, but that didn't make any differences.  I tried my updated 64-bit Windows 10 Pro PC's v4.3.3 and old Newsbin Pro v6.90. No problems! What's wrong with my new nzbget? I can't share these *.nzb files since they were from private servers.

Thank you for reading and hopefully answering soon. :)

**Comment by dnzbk** [comment]
Hello,

NZBGet looks for the password in the name of the nzb file in brackets like `{{some password}}` or in the nzb file itself in the `<meta type=“password”>` line. 
Otherwise, it's required to specify the password manually by clicking on the download's name -> Postprocess -> Password.
Are you sure the password from the nzb file or that you specified is correct?
If not, then you could try to unpack the files manually to make sure:
1) Temporarily disable the unpacking postprocess: download's name -> Postprocess
2) Wait for a full download
3) Go to the `${DestDir}` and try to unpack it manually using the password you have

**Comment by antdude** [comment]
Thank you for answering quickly. I looked inside in a nzb file (never knew it was a text file) that started with (changed its password a little just in case):
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE nzb PUBLIC "-//newzBin//DTD NZB 1.1//EN" "http://www.newzbin.com/DTD/nzb/nzb-1.1.dtd">
<nzb xmlns="http://www.newzbin.com/DTD/2003/nzb">
<head>
 <meta type="password">r6czTEhnOX5gZO286Ant</meta>
</head>
...
```

I went into my ~/Downloads/nzbget/intermediate/F.#1/ and manually tried with unrar command with the manually entered password shown in nzb file. It worked. Do I assume nzbget is supposed to automatically use this password to unrar? If so, then why is it failing to do it?

**Comment by dnzbk** [comment]
Yes, nzbget should automatically use this password to unrar.
I've just double-checked this and in my case it works fine. So, I'm not sure what's going on your use-case.
Is there any chance to share this nzb file with me (denis@nzbget.com)?
Could you send me at least your `nzbget.conf`? 
Without any private data, of course.

**Comment by antdude** [comment]
_link_was_here_ (will delete after you review it).

**Comment by dnzbk** [comment]
Thanks. 
Your config looks good to me. 
Unfortunatelly, I’m not sure what’s causing the issue. Hopefully I'll encounter a similar nzb and reproduce the issue.

**Comment by antdude** [comment]
dnzbk: I sent you an e-mail. ;)

**Comment by antdude** [comment]
FYI. dnzbk was able to reproduce and is fixing my issue with a NZB sample and only happens in command line.

**Comment by dnzbk** [comment, marked as answer]
@antdude 
The new build with the fix is [ready](https://github.com/nzbgetcom/nzbget/releases/tag/testing). 
Feel free to test it.

**Comment by antdude** [comment]
Just tested it since this was my bug! Perfect and thanks. Note: I reused my my old nzbget.conf file since I was lazy to reconfigure from scratch. :)
